ipaplatform/redhat/authconfig.py

```diff
@@ -18,15 +18,95 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+import logging
+import six
+import abc
+
 from ipaplatform.paths import paths
 from ipapython import ipautil
 from ipapython.admintool import ScriptError
 import os
 
 FILES_TO_NOT_BACKUP = ['passwd', 'group', 'shadow', 'gshadow']
 
+logger = logging.getLogger(__name__)
+
+
+@six.add_metaclass(abc.ABCMeta)
+class RedHatAuthTool(object):
+
+    @staticmethod
+    def get_tool():
+        return RedHatAuthSelect()
+
+    @abc.abstractmethod
+    def configure(self, sssd, mkhomedir, statestore):
+        pass
+
+    @abc.abstractmethod
+    def unconfigure(self, fstore, statestore,
+                    was_sssd_installed,
+                    was_sssd_configured):
+        pass
+
+    @abc.abstractmethod
+    def backup(self, path):
+        pass
+
+    @abc.abstractmethod
+    def restore(self, path):
+        pass
+
+
+class RedHatAuthSelect(RedHatAuthTool):
+
+    def configure(self, sssd, mkhomedir, statestore):
+        if not sssd:
+            logger.error("no-sssd options is not supported by installer. "
+                         "For client installation without sssd check the "
+                         "output of 'ipa-advise config-fedora-authconfig'")
+            return
+            # raise ScriptError("Non-sssd installation is not supported")
+
+        # Do not raise exception since authselect may fail just
+        # because there is PAM configuration already in place.
+        if mkhomedir:
+            ipautil.run(
+                [
+                    paths.AUTHSELECT,
+                    "select", "sssd",
+                    "with-mkhomedir",
+                    "--force"
+                ]
+            )
+        else:
+            ipautil.run(
+                [
+                    paths.AUTHSELECT,
+                    "select", "sssd",
+                    "--force"
+                ]
+            )
+
+    def unconfigure(self, fstore, statestore,
+                    was_sssd_installed,
+                    was_sssd_configured):
+        pass
+
+    def backup(self, path):
+        # not implemented in authselect
+        pass
+
+    def restore(self, path):
+        # not implemented in authselect
+        pass
```

Comment:
If backup and restore do nothing, I'm afraid that the following scenario would not restore the right sssd profile:
```
ipa-server-install
ipa-backup
ipa-server-install --uninstall -U
authselect select winbind (to select another profile)
ipa-restore <path_to_backup>
```
^^ at this point we probably end up with the winbind profile instead of sssd

We can probably save the profile and its enabled features (use ```authselect current``` and parse the output), and re-apply the same config during restore.

Reply:
@flo-renaud, thank you. Will fix this as you proposed.
I will have to implement backup/restore for authselect. This will solve both issues about restore which you raised. 

Authselect will have its own backup/restore functionality in future. It looks like we are running in front of a train :)

Reply:
@flo-renaud, I have just added backup/restore implemented in the same way as you advised.